src/main/java/org/synyx/urlaubsverwaltung/absence/web/AbsenceOverviewViewController.java

```diff
@@ -142,35 +142,52 @@ public String absenceOverview(
         model.addAttribute("selectedMonth", selectedMonth);
 
         final List<Person> membersOfSignedInUser = getActiveMembersOfPerson(signedInUser);
-        model.addAttribute("showRichLegend", !membersOfSignedInUser.isEmpty());
+        final boolean isSignedInUserAllowedToSeeAbsences = !membersOfSignedInUser.isEmpty();
+        model.addAttribute("sickNoteLegendVisible", isSignedInUserAllowedToSeeAbsences);
 
-        // use active vacation types instead of all to avoid too much items in the legend.
+        final List<VacationType> vacationTypes = vacationTypeService.getAllVacationTypes();
+        final Map<Integer, VacationType> vacationTypesById = vacationTypes.stream().collect(toMap(VacationType::getId, Function.identity()));
+
+        // use active vacation types instead of all to avoid too many items in the legend.
         // (there could be non-active vacation types visible in the absence-overview)
         // the legend will be removed soon -> therefore display just the active items
-        final List<VacationType> vacationTypes = vacationTypeService.getActiveVacationTypes();
-        model.addAttribute("vacationTypeColors", vacationTypes.stream().map(AbsenceOverviewViewController::toVacationTypeColorsDto).collect(toUnmodifiableList()));
+        List<VacationTypeColorDto> vacationTypeColorDtos = prepareVacationTypeColorsForLegend(isSignedInUserAllowedToSeeAbsences, vacationTypes);
+        model.addAttribute("vacationTypeColors", vacationTypeColorDtos);
+
+        // TODO this information should be part of the domain imho. (`AbsencePeriod.RecordInfo`)
```

Comment:
Moved the `visibleToEveryone` Information to RecordInfo, but not sure about the person-aware part (`!(membersOfSignedInUser.contains(recordInfo.getPerson())`). @bseber what do you thing?